### ISSUE TITLE
Domain literals break vim syntax highlighting

### ISSUE THREAD

**Issue by daviditen** [chapel-lang/chapel#11509]
When looking at a code that uses domain literals in vim with syntax highlighting, all curly braces after the domain literal are marked as errors (highlighted in red).

For example, in the code:
```chpl
proc f(d: domain) { }

proc main {
  f({1..5});

  for i in 1..5 {
  }
}
```

Every curly brace including and after {1..5} are highlighted in red.

![screen shot 2018-10-30 at 12 38 27 pm](https://user-images.githubusercontent.com/7740834/47738239-beafa580-dc40-11e8-8352-7550a8f86a4c.png)

**Comment by bradcray** [comment]
It doesn't happen very often, so I have to say it loud and proud:  

**Emacs mode doesn't have this problem!!**  :D
===================================

**Comment by daviditen** [comment]
Fixed by #12563.
